paasta_tools/deployd/watchers.py

```diff
@@ -61,27 +61,31 @@ def watch_folder(self, path):
             for child in children:
                 self.watch_folder("{}/{}".format(path, child))
 
+    def _make_service_instance(self, path):
+        service, instance = path.split('/')[-3:-1]
+        self.log.info("Number of instances changed or autoscaling enabled for first time"
+                      " for {}.{}".format(service, instance))
+        # https://github.com/python/mypy/issues/2852
+        service_instance = ServiceInstance(  # type: ignore
+            service=service,
+            instance=instance,
+            cluster=self.config.get_cluster(),
+            bounce_by=int(time.time()),
+            bounce_timers=None,
+            watcher=type(self).__name__,
+            failures=0,
+        )
+        self.inbox_q.put(service_instance)
+
     def watch_node(self, path):
-        self.log.debug("Adding watch on {}".format(path))
+        self.log.debug("Adding node watch on {}".format(path))
         DataWatch(self.zk, path, func=self.process_node_event, send_event=True)
+        self._make_service_instance(path)
```

Comment:
This sounds OK, but is won't this pick up on all random node watches, like changes to files that are not relevant?

Specifically, don't we only want to call this when the nodepath ends in "instances" ?

Reply:
Should be okay because we only call this from a block that does an if on "instances"

Reply:
I think the check has been done at line 58:  " if children and ('instances' in children) "